e2e/e2e_test.go

```diff
@@ -60,26 +59,34 @@ var _ = Describe("E2E tests", func() {
 				}
 			})
 
-			It("recreates the associated Machine", func() {
+			It("deletes the associated Machine", func() {
 				By("checking the Machine was deleted")
 				Eventually(func() bool {
 					key := client.ObjectKeyFromObject(machine)
 					err := k8sClient.Get(context.TODO(), key, createMachineStruct())
 					return errors.IsNotFound(err)
 				}, 5*time.Minute, 10*time.Second).Should(BeTrue())
 
-				By("checking the Node is recreated")
+				By("checking a new Node was created after the CR")
+				req, _ := labels.NewRequirement(workerLabelName, selection.Exists, []string{})
+				selector := labels.NewSelector().Add(*req)
+
+				newWorkers := &v1.NodeList{}
+				Eventually(func() int {
+					Expect(k8sClient.List(context.Background(), newWorkers, &client.ListOptions{LabelSelector: selector})).ToNot(HaveOccurred())
+					return len(newWorkers.Items)
+				}, 15*time.Minute, 10*time.Second).Should(BeNumerically("==", len(initialWorkers.Items)))
```

Comment:
I think this test can be flaky: at this point we only know that the machine was deleted. Depending on how fast the corresponding node is deleted and recreated, we might still see the old node here, not? Then the following test checking the creation date just fails. Suggestion: skip this test, and wrap the next one in a Eventually block. WDYT?

Reply:
I might have made a wrong assumption here. I thought that the Node was running "on" the Machine, and then it made sense to me that the latter is destroyed only after the Node, but if it is the other way around, wrapping the last one on an Eventually block is correct.

Reply:
confirmed that I've made the wrong assumption :smile: 
I will update the PR

Reply:
> I thought that the Node was running "on" the Machine

Technically both are just abstractions, some data in a key value store, so you can not really say something "runs on" another thing (yes I saw the quotes, just wanted to reiterate this, because its important).
The technical reason why the node is deleted some time after the machine: the node is deleted by the machine controller, triggered by machine deletion 🙂